scalr/resource_scalr_provider_configuration_default.go

```diff
@@ -0,0 +1,197 @@
+package scalr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scalr/go-scalr"
+)
+
+var resourceScalrProviderConfigurationDefaultMutex sync.Mutex
+
+func resourceScalrProviderConfigurationDefault() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalrProviderConfigurationDefaultCreate,
+		ReadContext:   resourceScalrProviderConfigurationDefaultRead,
+		DeleteContext: resourceScalrProviderConfigurationDefaultDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceScalrProviderConfigurationDefaultImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"provider_configuration_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"environment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceScalrProviderConfigurationDefaultImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	scalrClient := meta.(*scalr.Client)
+
+	id := d.Id()
+
+	providerConfiguration, environment, err := getPCDLinkedResources(ctx, id, scalrClient)
+	if err != nil {
+		if errors.Is(err, scalr.ErrResourceNotFound) {
+			return nil, fmt.Errorf("provider configuration default %s not found", id)
+		}
+		return nil, fmt.Errorf("error retrieving provider configuration default %s: %v", id, err)
+	}
+
+	_ = d.Set("provider_configuration_id", providerConfiguration.ID)
+	_ = d.Set("environment_id", environment.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceScalrProviderConfigurationDefaultCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	resourceScalrProviderConfigurationDefaultMutex.Lock()
```

Comment:
This should do the trick for now.
However the root cause is in the API design here, not provider.
May be we should consider having `/environment/<env-id>/relationships/default-provider-configurations` endpoints for granular defaults management.
cc @DayS1eeper